### PR TITLE
Add sample react native app

### LINKED ISSUE #22: Add sample app

### PR DESCRIPTION
**Pull request by axolotl0212**
closes #22

# Screenshot

Screenshot_20201105-163204_reactNativeExample
Screenshot 2020-11-05 at 4 46 07 PM

### CODE REVIEW

**Comment by axolotl0212** [comment]
rebased

File: reactNativeExample/ios/reactNativeExample.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -664,7 +663,7 @@
 					"-ObjC",
 					"-lc++",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = "org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)";
+				PRODUCT_BUNDLE_IDENTIFIER = com.authgear.reactNativeExample;
```
Comment (louischan-oursky) [inline reactNativeExample/ios/reactNativeExample.xcodeproj/project.pbxproj:666]: The bundle identifier should be in lowercase. I suggest using the ID `com.authgear.example.reactnative`.

File: reactNativeExample/.buckconfig [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (louischan-oursky) [inline reactNativeExample/.buckconfig:1]: Can we put the example app in `/example/reactnative`?

**Review by louischan-oursky — COMMENTED** [review]
1. If I logged in, next time when I open the app, logged in status is not restored.
2. After login, I cannot click any button.

**Comment by louischan-oursky** [comment]
It seems that we cannot run the ios app without any changes given that the port has been changed to 8082. I had to change the source code of react native to make the default one 8082. I guess we can make some changes to AppDelegate.m to tell them explicitly our new port?

File: reactNativeExample/ios/reactNativeExample.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -0,0 +1,969 @@
...
+			inputPaths = (
+			);
+			name = "Start Packager";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "export RCT_METRO_PORT=\"${RCT_METRO_PORT:=8082}\"\necho \"export RCT_METRO_PORT=${RCT_METRO_PORT}\" > \"${SRCROOT}/../node_modules/react-native/scripts/.packager.env\"\nif [ -z \"${RCT_NO_LAUNCH_PACKAGER+xxx}\" ] ; then\n  if nc -w 5 -z localhost ${RCT_METRO_PORT} ; then\n    if ! curl -s \"http://localhost:${RCT_METRO_PORT}/status\" | grep -q \"packager-status:running\" ; then\n      echo \"Port ${RCT_METRO_PORT} already in use, packager is either not running or not running correctly\"\n      exit 2\n    fi\n  else\n    open \"$SRCROOT/../node_modules/react-native/scripts/launchPackager.command\" || echo \"Can't start packager automatically\"\n  fi\nfi\n";
```
Comment (axolotl0212) [inline reactNativeExample/ios/reactNativeExample.xcodeproj/project.pbxproj:533]: I actually changed from port 8081 to 8082 here in iOS, is it not working? or you run the app from command line? @louischan-oursky

**Comment by axolotl0212** [comment]
> After login, I cannot click any button.

This is unexpected, will investigate

**Comment by axolotl0212** [comment]
> If I logged in, next time when I open the app, logged in status is not restored.

Will handle this with @react-native-community/async-storage. btw according to the [npm site](https://www.npmjs.com/package/@react-native-community/async-storage), seems like this is deprecated, maybe we need to update the react native setup doc?

**Comment by axolotl0212** [comment]
> After login, I cannot click any button.

Cannot reproduce this. is it iOS or Android? what is the OS version?

**Comment by louischan-oursky** [comment]
> Will handle this by attempting to get access token from storage. btw according to the npm site, seems like this is deprecated, maybe we need to update the react native setup doc?

Thank you for telling me about the deprecation notice. Let's update our peer dependency and dev dependency to depend on the new `@react-native-async-storage/async-storage: ^1.0.0`

> Cannot reproduce this. is it iOS or Android? what is the OS version?

iOS simulator of iPhone SE 2 iOS 13

File: reactNativeExample/ios/reactNativeExample.xcodeproj/project.pbxproj (hunk 1 above)
Comment (axolotl0212) [reply reactNativeExample/ios/reactNativeExample.xcodeproj/project.pbxproj]: changed

File: reactNativeExample/.buckconfig (hunk 2 above)
Comment (axolotl0212) [reply reactNativeExample/.buckconfig]: moved to /example/reactnative

**Comment by axolotl0212** [comment]
@louischan-oursky updated, will try to reproduce the bug you mentioned above (After login, I cannot click any button.)

**Comment by axolotl0212** [comment]
@louischan-oursky reproduced by simulator. Found an issue (dialog is shown on top of loading modal). Replaced with a view without modal

**Comment by louischan-oursky** [comment]
Have you encountered this error?

```
$ pod install
Auto-linking React Native modules for target `reactNativeExample`: RNCAsyncStorage, authgear-react-native, and react-native-config
Analyzing dependencies
[!] Unable to find a target named `reactNativeExample-tvOS` in project `reactNativeExample.xcodeproj`, did find `reactNativeExample` and `reactNativeExampleTests`.
```

I saw you deleted something similar. Is that related?

**Comment by axolotl0212** [comment]
> I saw you deleted something similar. Is that related?

I removed the tvOS target, but I didn't remove the corresponding code in Podfile as I didn't run pod install (no new dependencies). Will remove this and the error should be resolved

**Comment by axolotl0212** [comment]
Fixed @louischan-oursky

**Comment by louischan-oursky** [comment]
In AppDelegate, we can tell the RN we have changed the packager port by

```
- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  RCTBundleURLProvider *settings = [RCTBundleURLProvider sharedSettings];
  settings.jsLocation = "localhost:8082";
  return [settngs jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}
```

**Comment by axolotl0212** [comment]
Will add this to app delegate

**Comment by axolotl0212** [comment]
@louischan-oursky updated

**Pull request merged**